packages/web-components/src/components/date-picker/date-picker.scss

```
@@ -77,12 +77,24 @@ $css--plex: true !default;
     display: none;
   }
 }
-:host(#{$prefix}-date-picker-input[warn]) {
+:host(#{$prefix}-date-picker-input[warn]:not([disabled]):not([readonly])) {
```

```suggestion
:host(#{$prefix}-date-picker-input[warn]:not([invalid])) {
```

The warn-text doesn't render when disabled/readonly are true so this isn't really needed. It would be better to do `:not[(invalid)]` so that this colour does not show on the invalid-text if warn and invalid are both true